### ISSUE TITLE
Server doesn't exit the process after shutting down

### ISSUE THREAD

**Issue by Solly-W** [StrangeLoopGames/EcoIssues#12020]
Running a server on Ubuntu 18.04, Eco version 0.8.1.2.
After shutting down, the process doesn't exit, and doesn't respond to a SIGTERM. This happens every time.
[Recording of Terminal](https://github.com/StrangeLoopGames/EcoIssues/files/3077256/Recording.of.Terminal.zip)

Log:
```[17:11:59.970] [1] [ Info] [Eco] >>> START EcoServer, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null 
[17:11:59.996] [1] [ Info] [Eco] Eco Server 0.8.1.2 beta
 
[17:12:00.060] [1] [ Info] [Eco] Server Initialization...  
[17:12:00.060] [1] [ Info] [Eco] Starting LocalizationPlugin...  
[17:12:00.327] [1] [ Info] [Eco] Starting ModKitPlugin...  
[17:12:00.327] [1] [ Info] [Eco] Loading mods...  
[17:12:00.385] [1] [ Info] [Eco] Compiling mods...  
[17:12:02.199] [1] [ Info] [Eco] Loading Eco.IntegrationTests...
 
[17:12:02.204] [1] [ Info] [Eco] Loading mods finished in 1,876.0ms
 
[17:12:02.204] [1] [ Info] [Eco] Starting NetworkManager...  
[17:12:02.210] [5] [ Info] [Eco] Initializing Simulation Types...  
[17:12:02.358] [7] [ Info] [Eco] Starting BackupPlugin...  
[17:12:02.369] [7] [ Info] [Eco] Starting StorageManager...  
[17:12:02.488] [8] [ Info] [Eco] Starting WorldGeneratorPlugin...  
[17:12:02.490] [6] [ Info] [Eco] Starting GameplayStatsPlugin...  
[17:12:02.842] [5] [ Info] [Eco] Initializing items...  
[17:12:06.214] [7] [ Info] [Eco] Starting PausePlugin...  
[17:12:06.218] [7] [ Info] [Eco] Loading Pause...  
[17:12:08.930] [5] [ Info] [Eco] Initializing skills...  
[17:12:09.050] [7] [ Info] [Eco] Starting StatDatabase...  
[17:12:09.050] [9] [ Info] [Eco] Starting WorldPlugin...  
[17:12:09.091] [9] [ Info] [Eco] Loading World...  
[17:12:16.716] [9] [ Info] [Eco] Initializing world...  
[17:12:17.707] [9] [ Info] [Eco] Starting TimePlugin...  
[17:12:17.707] [9] [ Info] [Eco] Loading Time...  
[17:12:17.707] [9] [ Info] [Eco] Starting ReputationPlugin...  
[17:12:17.707] [9] [ Info] [Eco] Loading Reputation...  
[17:12:17.711] [9] [ Info] [Eco] Starting PlayerHousingManager...  
[17:12:17.711] [9] [ Info] [Eco] Starting UserManager...  
[17:12:17.725] [9] [ Info] [Eco] Loading Users...  
[17:12:17.842] [9] [ Info] [Eco] Initializing inventories...  
[17:12:17.868] [9] [ Info] [Eco] Starting EcoObjectManager...  
[17:12:17.869] [13] [ Info] [Eco] Starting ControllerManager...  
[17:12:17.869] [12] [ Info] [Eco] Starting ChatServer...  
[17:12:17.869] [16] [ Info] [Eco] Starting GarbagePlugin...  
[17:12:17.869] [15] [ Info] [Eco] Starting DifficultySettings...  
[17:12:17.869] [11] [ Info] [Eco] Starting DisasterPlugin...  
[17:12:17.869] [14] [ Info] [Eco] Starting EconomyPlugin...  
[17:12:17.869] [16] [ Info] [Eco] Loading Garbage...  
[17:12:17.874] [14] [ Info] [Eco] Loading Economy...  
[17:12:17.981] [14] [ Info] [Eco] Starting Legislation...  
[17:12:17.981] [11] [ Info] [Eco] Loading Disasters...  
[17:12:17.981] [14] [ Info] [Eco] Loading Laws...  
[17:12:18.193] [14] [ Info] [Eco] Starting PropertyManager...  
[17:12:18.193] [12] [ Info] [Eco] Loading Chat...  
[17:12:18.200] [14] [ Info] [Eco] Loading Property...  
[17:12:18.210] [14] [ Info] [Eco] Starting MinimapManager...  
[17:12:18.210] [14] [ Info] [Eco] Starting EcoSim...  
[17:12:18.245] [14] [ Info] [Eco] Loading Plants...  
[17:12:19.473] [14] [ Info] [Eco] Initializing plants...  
[17:12:20.443] [14] [ Info] [Eco] Starting RouteProbingPlugin...  
[17:12:20.443] [14] [ Info] [Eco] Loading RouteProbing...  
[17:12:22.385] [14] [ Info] [Eco] Starting WorldLayerManager...  
[17:12:22.385] [14] [ Info] [Eco] Loading Layers...  
[17:12:22.540] [14] [ Info] [Eco] Initializing world layers...  
[17:12:22.540] [14] [ Info] [Eco] Loading ClimateSim...  
[17:12:23.589] [14] [ Info] [Eco] Starting EconomyTracker...  
[17:12:23.591] [14] [ Info] [Eco] Starting UserTracker...  
[17:12:23.591] [14] [ Info] [Eco] Starting WorldObjectManager...  
[17:12:23.672] [14] [ Info] [Eco] Loading WorldObjects...  
[17:12:23.687] [14] [ Info] [Eco] Starting RoomPlugin...  
[17:12:23.687] [14] [ Info] [Eco] Loading Rooms...  
[17:12:23.693] [14] [ Info] [Eco] Starting ObjectivePlugin...  
[17:12:23.693] [14] [ Info] [Eco] Loading Objectives...  
[17:12:23.698] [14] [ Info] [Eco] Starting SleepPlugin...  
[17:12:23.698] [14] [ Info] [Eco] Loading Sleep...  
[17:12:23.701] [1] [ Info] [Eco] Starting WebServerPlugin...  
[17:12:23.709] [1] [ Info] [Eco] Starting OtherAliasCategory...  
[17:12:23.713] [9] [ Info] [Eco] Starting IntegrationTestsPlugin...  
[17:12:23.727] [1] [ Info] [Eco] Initializing LocalizationPlugin...  
[17:12:23.787] [1] [ Info] [Eco] Initializing ModKitPlugin...  
[17:12:23.794] [1] [ Info] [Eco] Initializing WorldGeneratorPlugin...  
[17:12:23.794] [1] [ Info] [Eco] Initializing UserManager...  
[17:12:23.815] [1] [ Info] [Eco] Initializing GarbagePlugin...  
[17:12:23.817] [1] [ Info] [Eco] Initializing EconomyPlugin...  
[17:12:23.825] [1] [ Info] [Eco] Initializing Legislation...  
[17:12:24.165] [1] [ Info] [Eco] Initializing PropertyManager...  
[17:12:24.168] [1] [ Info] [Eco] Initializing MinimapManager...  
[17:12:24.476] [1] [ Info] [Eco] Initializing UserTracker...  
[17:12:24.480] [1] [ Info] [Eco] Initializing WorldObjectManager...  
[17:12:24.525] [1] [ Info] [Eco] Initializing RoomPlugin...  
[17:12:24.525] [1] [ Info] [Eco] Initializing SleepPlugin...  
[17:12:24.526] [1] [ Info] [Eco] Initializing WorldPlugin...  
[17:12:24.526] [1] [ Info] [Eco] Ticking...  
[17:12:28.702] [1] [ Info] [Eco] Compressing...  
[17:12:30.807] [1] [ Info] [Eco] Initializing NetworkManager...  
[17:12:30.841] [1] [ Info] [Eco] Initializing WorldLayerManager...  
[17:12:30.841] [1] [ Info] [Eco] Initializing WebServerPlugin...  
[17:12:30.848] [1] [ Info] [Eco] Skipping Windows firewall setup, platform = Unix

[17:12:30.848] [1] [ Info] [Eco] (You may need to manually open port 3001 if the EcoWebServer is inaccessible)
 
[17:12:31.115] [1] [ Info] [Eco] Initializing IntegrationTestsPlugin...  
[17:12:31.131] [1] [ Info] [Eco] Server Initialization finished in 31,070.3ms
 
[17:12:31.295] [79] [ Info] [Eco] Skipping Windows firewall setup, platform = Unix
 
[17:12:31.295] [79] [ Info] [Eco] (You may need to manually open port 3000 if the EcoGameServer is inaccessible)
 
[18:19:23.590] [105] [ Info] [Eco] Shutdown signal received
 
[18:19:23.590] [105] [ Info] [Eco] Saving...
 
[18:19:23.592] [105] [ Info] [Eco] Saving world...  
[18:19:24.427] [105] [ Info] [Eco] Done
 
[18:19:24.760] [105] [ Info] [Eco] Shutdown finished
 ```

**Comment by D3nnis3n** [comment]
Low Priority, because we don't officially support linux servers (yet).

**Comment by ky0uraku** [comment]
Same behaivor sometimes happen on our windows server - just hang after "Saving world...Done" message and not close any of window (console/admin-gui).

**Comment by mirasrael** [comment]
Should be fixed in 8.2
